File: PI/CPP/preprocessing.cpp
#include "preprocessing.hpp"

Preprocessing::Preprocessing(int argc, char **argv) {
    if (argc != 3) {
        std::cerr << "Usage: " << argv[0] << " <number of iterations> <number of threads>";
        exit(0);
    }

    m_nb_iterations = atoi(argv[1]);
    m_nb_threads = atoi(argv[2]);
}

void Preprocessing::timer_start() {
    m_start = std::chrono::high_resolution_clock::now();
}

void Preprocessing::timer_stop() {
    m_end = std::chrono::high_resolution_clock::now();
}

double Preprocessing::timer_duration() const {
    return std::chrono::duration_cast<std::chrono::microseconds>(m_end - m_start).count() * 1e-6;
}

size_t Preprocessing::get_nb_threads() const { return m_nb_threads; }

size_t Preprocessing::get_nb_iterations() const { return m_nb_iterations; }

File: PI/Java/src/main/java/Preprocessing.java
public class Preprocessing {
    private int nbIterations;
    private int nbThreads;
    private long startTime;
    private long endTime;

    public Preprocessing(String[] args) {
        if (args.length != 2) {
            throw new RuntimeException("Usage: <number of iterations> <number of threads>");
        }

        nbIterations = Integer.parseInt(args[1]);
        nbThreads = Integer.parseInt(args[1]);
    }

    public void timerStart() {
        startTime = System.nanoTime();
    }

    public void timerStop() {
        endTime = System.nanoTime();
    }

    public double timerDuration() {
        return (endTime - startTime) * 1.0e-9;
    }

    public int getNbIterations() {
        return nbIterations;
    }

    public int getNbThreads() {
        return nbThreads;
    }
}
